Language/Libraries_and_Scripts/Imports/Semantics_of_Imports/same_uri_A01_t10.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion All imports and exports of the same URI in a Dart program denotes
+/// the same library, and imports or exports of different URIs denote distinct
+/// libraries.
+///
+/// @description Checks that imports of the same URI in a Dart program denotes
+/// the same library.
+/// @author sgrekhov22@gmail.com
+
+import '../../../../Utils/expect.dart';
+import 'same_uri_common_lib.dart' as l1;
+import '../Semantics_of_Imports/same_uri_common_lib.dart' as l2;
```

Comment:
I wouldn't expect these to be the same URI. Doesn't the test fail?

Reply:
OK, the discussion in https://github.com/dart-lang/sdk/issues/56869 revealed that at least some tools do already normalize the URIs to some extent.

The language specification does say that "different" URIs denote different libraries, and "equal" URIs denote the same library, so we should be able to test that.

Reply:
> I wouldn't expect these to be the same URI. Doesn't the test fail?
No, the test works well in both analyzer and vm on Win and Linux.

Reply:
Right, I found a comment to the effect that at least some tools choose to simplify paths where `..` occurs in some non-first location (so `SOMETHING/dir1/dir2/../dir3/foo.dart` is simplified to `SOMETHING/dir1/dir3/foo.dart`). The tools can in principle have any implementation specific behavior they want in almost every respect when it comes to URIs, so we should just test this in a minimal fashion and leave the details to the tool teams and their specialized test suite.